kernel/src/debug.rs

```diff
@@ -343,10 +343,12 @@ pub struct DebugWriter {
 /// needed so the debug!() macros have a reference to the object to use.
 static mut DEBUG_WRITER: Option<&'static mut DebugWriterWrapper> = None;
 
+pub unsafe fn try_get_debug_writer() -> Option<&'static mut DebugWriterWrapper> {
```

Comment:
I don't think this should be a `pub` function. I see that this mirrors the non-try version, but that really should not be `pub` either.. 

The underlying `DEBUG_WRITER` is pretty unsafe (and not a `pub` object). Constraining its access to this file seems valuable.

--

Could you remove `pub` here and while we're at it from the non-try variant?

Reply:
Can do :+1:.

Reply:
Thanks for the hint. I've pushed the changes and added the following to the commit message:

_Due to the static mutable debug writer being inherently unsafe, this also reduces its visibility._

Feel free to resolve this if you are comfortable with the result